src/MessageHandler/DatasetSubmissionFilerHandler.php

```diff
@@ -153,12 +154,13 @@ public function __invoke(DatasetSubmissionFiler $datasetSubmissionFiler)
     /**
      * Method to process a single file.
      *
-     * @param File  $file           The file that is being processed.
-     * @param array $loggingContext The logging context for the related dataset submission.
+     * @param File   $file           The file that is being processed.
+     * @param array  $loggingContext The logging context for the related dataset submission.
+     * @param string $udi            The UDI of the dataset.
      *
      * @return void
      */
-    private function processFile(File $file, array $loggingContext): void
+    private function processFile(File $file, array $loggingContext, string $udi): void
```

Comment:
UDI should probably be second. I'm thinking most significant parameters first. Logging context could even be empty.